python/pyspark/sql/dataframe.py

```diff
@@ -2188,6 +2188,142 @@ def cube(self, *cols: "ColumnOrName") -> "GroupedData":  # type: ignore[misc]
 
         return GroupedData(jgd, self)
 
+    def unpivot(
+        self,
+        ids: Optional[Union["ColumnOrName", List["ColumnOrName"], Tuple["ColumnOrName", ...]]] = None,
+        values: Optional[Union["ColumnOrName", List["ColumnOrName"], Tuple["ColumnOrName", ...]]] = None,
```

Comment:
What are your thoughts on allowing for tuple of columns as an alternative to list of columns (equivalent semantics)? `Dataframe._jcols` does not support tuples, only list and single string so this does not seem to be standard behaviour.

Reply:
I'm netural on it.

some existing methods also use tuple of column names:

```
    def freqItems(
        self, cols: Union[List[str], Tuple[str]], support: Optional[float] = None
    ) -> "DataFrame":
```